### PR TITLE
The Emitter now emit body in parts

### PR DESCRIPTION
**Pull request by roxblnfk**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Tests pass?   | ✔️
| Fixed issues  | #165

### CODE REVIEW

File: src/Emitter/SapiEmitter.php [hunk 1]
```diff
@@ -10,6 +10,12 @@
 final class SapiEmitter implements EmitterInterface
 {
     private const NO_BODY_RESPONSE_CODES = [204, 205, 304];
+    private $partSize;
+
+    public function __construct(int $partSize = 8388608)
```
Comment (samdark) [inline src/Emitter/SapiEmitter.php]: What is 8388608? Worth moving to a constant such as `private DEFAULT_BUFFER_SIZE` and adding phpdoc saying why it is exactly 8 megabytes.

**Comment by samdark** [comment]
Merged. Thank you!

File: src/Emitter/SapiEmitter.php [hunk 2]
```diff
@@ -10,6 +10,13 @@
 final class SapiEmitter implements EmitterInterface
 {
     private const NO_BODY_RESPONSE_CODES = [204, 205, 304];
+    private $bufferSize;
+    private const DEFAULT_BUFFER_SIZE = 8388608; // 8MB
+
+    public function __construct(int $bufferSize = self::DEFAULT_BUFFER_SIZE)
+    {
+        $this->bufferSize = $bufferSize > 0 ? $bufferSize : self::DEFAULT_BUFFER_SIZE;
```
Comment (kamarton) [inline src/Emitter/SapiEmitter.php]: If zero or negative, the default approach is not really good. I suggests
- `?int .. = null` for parameter
- if negative or zero, throw InvalidParameterException
- if null then default.

File: src/Emitter/SapiEmitter.php [hunk 3]
```diff
@@ -10,6 +10,13 @@
 final class SapiEmitter implements EmitterInterface
 {
     private const NO_BODY_RESPONSE_CODES = [204, 205, 304];
+    private $bufferSize;
+    private const DEFAULT_BUFFER_SIZE = 8388608; // 8MB
```
Comment (rustamwin) [inline src/Emitter/SapiEmitter.php]: @samdark Do we have a clear order of properties? It looks a little weird

Comment (samdark) [reply src/Emitter/SapiEmitter.php]: Oh, right. Constants first.

Comment (kamarton) [reply src/Emitter/SapiEmitter.php]: Is it possible to create tests for this like changelog.md in yii2?

Comment (samdark) [reply src/Emitter/SapiEmitter.php]: Yes, it is possible.

Comment (samdark) [reply src/Emitter/SapiEmitter.php]: For now I've updated https://github.com/yiisoft/docs/blob/master/010-code-style.md#property-constant-and-method-placement to be clear about this case.

**Pull request merged**